src/schema/artists.js

```diff
@@ -22,8 +25,18 @@ const Artists = {
     },
     sort: ArtistSorts,
   },
-  resolve: (root, options, _request, { rootValue: { artistsLoader } }) =>
-    artistsLoader(options),
+  resolve: (
+    root,
+    options,
+    _request,
+    { rootValue: { artistLoader, artistsLoader } }
+  ) => {
+    if (options.slugs) {
+      return Promise.all(options.slugs.map(slug => artistLoader(slug)))
```

Comment:
Does this end up batching still? If not, then that should probably be considered and documented in the description.

Reply:
Similar question: would this trigger N API requests per article-view, where N artists are mentioned in the article? (My understanding is that--even with the caching loader--additional requests are still made to refresh the cache.)

That could be a very significant increase in API throughput (especially considering traffic to articles), and possibly unnecessary. These results could be cached aggressively.

Reply:
@alloy @joeyAghion Not how it's currently implemented, I can leave a comment to make that clear. 

I was actually about to add support fetching multiple artists by slugs in gravity but decided to not preemptively optimize based on the usage (https://github.com/artsy/gravity/pull/11705). After looking more deeply in potential server load from articles, we only have about 5 artist links per article on average. I'm not sure on the exact strategy `DataLoader` uses to refresh its cache but my understanding is we cache payloads in redis. (@alloy if you can clarify)

Also another reason I thought it'd be fine is we use that exact same code for trending artists which is a query used pretty often  (https://github.com/artsy/metaphysics/blob/c43bb86bc7354d1f42fc162491b20995b3861ba7/src/schema/artists/trending.js#L25).

Reply:
Those docs are about batching for the same item being asked for multiple times in one request. I **think** we'd only be caching the result per-artist once then re-requesting straight after, see: https://github.com/artsy/metaphysics/blob/master/src/lib/loaders/api/loader_without_authentication_factory.js#L45-L84

I believe this would result in 5 unique artist HTTP requests per article load

Reply:
We also have a Gravity API that can take a list of id's, `artistsLoader`, so the `Promise.all` linked above can just be replaced w/ that one call (`artistsLoader({ids})`).

However, that only works with mongo id's, and not slugs. Not to block this, but it might be nice to just always use mongo id's (rather than slugs, which can change), when referencing Gravity data (so a migration and work needed in Positron to go from slugs -> id's).

In the meantime, each of these would likely return data immediately from the cache, but each would trigger a cache refresh (so you'd still see those calls coming thru to Gravity).

In terms of updating Gravity to support a single call being able to return artists based on a list of slugs (analogous to the current `artistsLoader({ids})`), that's certainly possible (already started in https://github.com/artsy/gravity/pull/11705), but seems like a workaround/premature optimization. Instead of spending some effort there, it may make more sense for a bit of effort to switch away from slugs and to id's (if possible).

Reply:
@mzikherman using gravity `id` will not work in this instance because we are fetching based on slugs scraped from article text.  We are also implementing this on a few other models including `genes`.

Reply:
@mzikherman We've considered making the change you're suggesting in positron,  unfortunately, this data is not always available. It would require our editors to always tag artist links with the ID. this will be made client-side from the article component where we don't have access to ids, we have to go by URLs extracted from the rendered HTML.

Reply:
Yeah ~5 API requests per article view is what I was worried about, given my understanding that articles are a huge proportion of our traffic. Also, they're static. In order of preference:

1. It would be a million times more efficient (not exaggerating) to store the association with the artist's id. Slugs can change also, while ids shouldn't.
2. I'm pretty sure there are other examples of longer caching in MP. It shouldn't be hard to borrow that pattern here and cut off that very high volume of requests by responding from a cache of 24 hours or more. (The slug->id association is guaranteed to never change, though the other artist data in the response might benefit from freshness.)
3. Could the artist data be requested on-demand (i.e., on-hover or whatever the equivalent mobile behavior is) rather than at page-load?
4. Updating the Gravity endpoint to support look-ups by slug is not unreasonable if slugs are a must-have. Will comment there next.